### PR TITLE
fixed city nav

### PR DESCRIPTION
**Pull request by gcziprusz**
Once you scroll past the city nav it's position becomes fixed. 

screen shot 2015-06-30 at 5 05 35 pm

### CODE REVIEW

File: ui/components/city_nav/index.js [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
-        'scrollTop': $target.offset().top
-    }, 500, 'swing');
+  $("html, body").stop().animate({
+    scrollTop: $target.offset().bottom
+  }, 500, "swing");
 });
+
+var $window = $(window),
+$cityNav = $(".js-city-nav"),
+cityNavTop = $cityNav.offset().top,
```
Comment (finnursig) [inline ui/components/city_nav/index.js]: You should reset this when the user resizes its browser, else it triggers on wrong position.

File: ui/components/city_nav/index.js [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+fixedState = false,
+fixedCityNav = $cityNav.clone(true)
+                .addClass("city-nav--fixed city-nav--hidden")
+                .appendTo("body");
+
+$window.scroll(debounce(() => {
+  if ($window.scrollTop() > cityNavTop){
+    if(!fixedState){
+      fixedCityNav.removeClass("city-nav--hidden");
```
Comment (finnursig) [inline ui/components/city_nav/index.js]: I would 'detach' from dom when not active, it helps keeping the css low, makes and also looks nice in the devtool !  ;)

Also you could then skip the .clone() until its needed. (first scroll trigger)

**Comment by gcziprusz** [comment]
@finnursig good call on recalculating nav.top on resize.

File: ui/components/city_nav/index.js [hunk 3]
```diff
@@ -1,12 +1,43 @@
...
+    if(fixedState){
+      fixedCityNav.detach();
+      fixedState = false;
+    }
+  }
+}));
+
+$window.resize(debounce(() => {
+  cityNavTop = $cityNav.offset().top;
+}));
```
Comment (jcreamer898) [inline ui/components/city_nav/index.js]: You need to pass a debounce number here too... 

```js
+$window.resize(debounce(() => {
+  cityNavTop = $cityNav.offset().top;
+}, 100));
```

**Comment by gcziprusz** [comment]
@jcreamer898  thanks, anything above 10 creates jenkyfication

**Comment by jcreamer898** [comment]
Dooo it
